syntax = "proto3";

service TicTacToe {
  rpc SendGreeting(GreetingMessage) returns (GreetingResponse) {}
  rpc StartElection(ElectionMessage) returns (ElectionResult) {}
  rpc TicTacToe(TicTacToeRequest) returns (TicTacToeResponse) {}
}

message GreetingMessage {
  int32 sender_id = 1;
  string message = 2;
}

message GreetingResponse {
  int32 responder_id = 1;
  string message = 2;
  bool success = 3;
}

message ElectionMessage {
  int32 sender_id = 1;
  int32 election_id = 2;
}

message ElectionResult {
  int32 leader_id = 1;
  bool success = 2;
}

message TicTacToeRequest {
    string command = 1;
}

message TicTacToeResponse {
    string response = 1;
    repeated string board = 2;
}
